scenarios/deployonly/deploy/deploy_application.go

```diff
@@ -31,19 +33,25 @@ func Application(env envUtil.Env, suiteName string) (bool, error) {
 	_, err := application.Deploy(env, appName, toEnvironment)
 	if err != nil {
 		logger.Errorf("Error calling Deploy for application %s:  %v", appName, err)
-		return false, err
+		return err
 	}
 
 	// Get job
-	ok, jobSummary := test.WaitForCheckFuncOrTimeout(env, func(env envUtil.Env) (bool, interface{}) {
-		return job.IsListedWithStatus(env, config.App3Name, "Succeeded")
+	jobSummary, err := test.WaitForCheckFuncWithValueOrTimeout(env, func(env envUtil.Env) (*models.JobSummary, error) {
+		jobSummary, err := job.IsListedWithStatus(env, config.App3Name, "Succeeded")
+		if err != nil {
+			return nil, err
+		}
+		if jobSummary == nil {
+			return nil, fmt.Errorf("could not get listed job for application %s status '%s' - exiting", config.App3Name, "Succeeded")
+		}
```

Comment:
Should this if-block be moved out of the WaitForCheckFuncWithValueOrTimeout call? If a job reports Succeeded, but JobSummary is nil, there is no reason to retry the calls, and the Application function should just return with an error